src/machines/java/com/enderio/machines/common/blockentity/task/SpawnTask.java

```diff
@@ -20,23 +25,30 @@ public class SpawnTask extends PoweredTask{
 
     public static final int spawnTries = 10;
     private boolean complete;
-    private int energyCost = 40000;//TODO Custom Config/Json File
+    private int energyCost;
     private int energyConsumed = 0;
     private final PoweredSpawnerBlockEntity blockEntity;
     private float efficiency = 1;
+    private SpawnType spawnType = MachinesConfig.COMMON.SPAWN_TYPE.get();
+    @Nullable
+    private EntityType<? extends Entity> entityType;
 
     /**
      * Create a new powered task.
      *
      * @param energyStorage The energy storage used to power the task.
      */
-    public SpawnTask(PoweredSpawnerBlockEntity blockEntity, IMachineEnergyStorage energyStorage) {
+    public SpawnTask(PoweredSpawnerBlockEntity blockEntity, IMachineEnergyStorage energyStorage, Optional<ResourceLocation> rl) {
         super(energyStorage);
         this.blockEntity = blockEntity;
+        loadSoulData(rl);
     }
 
     @Override
     public void tick() {
+        if (entityType == null) {
+            return;
```

Comment:
Ensure that if this is the case, the machine doesn't softlock due to the job not concluding.

Reply:
Oh good catch!

Reply:
would simply setting completed as true be enough here? That should end the task.

Reply:
Yeah set completed to true to end the task